pkg/inject/inject.go

```diff
@@ -202,16 +203,10 @@ func (conf *ResourceConfig) GetOwnerRef() *metav1.OwnerReference {
 // from the namespace they belong to. If the namespace has a valid config key
 // that the pod does not, then it is appended to the pod's template
 func (conf *ResourceConfig) AppendNamespaceAnnotations() {
-	for _, key := range ProxyAnnotations {
-		if _, found := conf.nsAnnotations[key]; !found {
-			continue
-		}
-		if val, ok := conf.GetConfigAnnotation(key); ok {
-			conf.AppendPodAnnotation(key, val)
-		}
-	}
+	annotations := append(ProxyAnnotations, ProxyAlphaConfigAnnotations...)
+	annotations = append(annotations, k8s.ProxyInjectAnnotation)
```

Comment:
What's the reasoning for adding ProxyInjectAnnotation here instead of just adding it to ProxyAnnotations?  Are there places where ProxyAnnotations is used where ProxyInjectAnnotation should not be included?

Reply:
`ProxyAnnotations` is the list of annotations prepended with `config.linkerd.io`, which are used as labels in the metrics `proxy_inject_admission_requests_total` and `proxy_inject_admission_responses_total`. There's some mutation in the `validProxyConfigurationLabel` function that assumes the presence of that prefix, but we can make that more flexible if we decide to include `ProxyInjectAnnotation` as a label as well. LMKWYT.